hawkbit-autoconfigure/src/main/java/org/eclipse/hawkbit/autoconfigure/security/InMemoryUserManagementAutoConfiguration.java

```diff
@@ -93,13 +93,28 @@ UserDetailsService userDetailsService() {
         if (userPrincipals.isEmpty()) {
             final String name = securityProperties.getUser().getName();
             final String password = securityProperties.getUser().getPassword();
-            userPrincipals.add(new UserPrincipal(name, password, name, name, name, null, DEFAULT_TENANT,
-                    PermissionUtils.createAllAuthorityList()));
+            List<GrantedAuthority> authorityList;
+            if (securityProperties.getUser().getRoles().isEmpty()) {
+                authorityList = PermissionUtils.createAllAuthorityList();
+            } else {
+                authorityList = createAuthoritiesFromList(securityProperties.getUser().getRoles());
+            }
+            userPrincipals
+                    .add(new UserPrincipal(name, password, name, name, name, null, DEFAULT_TENANT, authorityList));
         }
 
         return new FixedInMemoryUserPrincipalUserDetailsService(userPrincipals);
     }
 
+    private static List<GrantedAuthority> createAuthoritiesFromList(final List<String> userAuthorities) {
+        List<GrantedAuthority> grantedAuthorityList = new ArrayList<>(userAuthorities.size());
```

Comment:
This looks almost the same as above (lines 78-82) can't you reuse same method there somehow as well?

Reply:
Isn't it the case that 'previous' lines 78-82 were refactored to be this method?

Reply:
@herdt-michael was really fast by doing this